bin/print_build_env.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python
+
+from string import Template
+import subprocess
+import os
+import re
+import contextlib
+
+##
+# Prints some details about the current build environment.
+#
+# Example Usage:
+#   ./print_build_env.py
+#
+
+BIN_DIR = os.path.dirname(__file__)
+webrtc_src_dir = os.path.join(BIN_DIR, '../webrtc/src')
+
+@contextlib.contextmanager
+def pushd(new_dir):
+    previous_dir = os.getcwd()
+    os.chdir(new_dir)
+    yield
+    os.chdir(previous_dir)
+
+def determine_git_branch(directory):
+    with pushd(directory):
+        git_branch_output = subprocess.check_output(["git", "branch"])
+        git_branch = [line.replace("* ", "") for line in git_branch_output.split("\n") if re.search("^\*", line)][0]
+        return git_branch
+
+def get_build_details():
+    template = Template("""## WebRTC Build Details
+
+To track down potential future issues, we log some of our build environment details.
+
+git branch:
+$git_branch
+
+xcode-select -p:
+$xcode_path
+
+hostname:
+$hostname
+
+gcc -v:
+$gcc_version
+""")
```

Comment:
os x version too?

Reply:
done via sw_vers. Final out put looks like this:
```
## WebRTC Build Details

To track down potential future issues, we log some of our build environment details.

webrtc git branch:
(HEAD detached at branch-heads/63)

webrtc git sha:
dc5904661c4c8b357e19595d80b2881db3f4a71b

build_script git sha:
6847d7ada965e930f57880274ef3b1abe2bf01dd

xcodebuild -version:
Xcode 9.2
Build version 9C40b

xcode-select -p:
/Applications/Xcode9.2.app/Contents/Developer

gcc -v:
Configured with: --prefix=/Applications/Xcode9.2.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 9.0.0 (clang-900.0.39.2)
Target: x86_64-apple-darwin16.7.0
Thread model: posix
InstalledDir: /Applications/Xcode9.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin

osx_version_details:
ProductName:    Mac OS X
ProductVersion: 10.12.6
BuildVersion:   16G1114

hostname:
ash.local
```